### DISCUSSION TITLE
No prompt to save scripts on closing

### DISCUSSION THREAD

**Discussion by doughobbs** [dbeaver/dbeaver#35237] [Q&A] [answered]
### Description

Previously, when closing a locally saved sql script or when closing the app I was prompted if I wanted save save it or not  - this behavior has now changed in the latest release (or one of the recent ones) so that it just closes without the prompt, this means that whenever I close things its auto-saving so my scripts are never a known state.

There is no auto-save option I can see in the preferences

Using version 24.1.4.202408041450 (Mac)

### DBeaver Version

24.1.4.202408041450

### Operating System

Mac 14.6.1 (23G93)

### Steps to reproduce

1. Open DBeaver
2. Open pre-saved sql script/s
3. Amend and use script/s as needed
4. Close script or application (no save prompt)
5. Re-open DBeaver
6. Re-open script from before - any changes previously made are there

**Comment by asmith04** [comment]
I also get the exact same problem. I have a project full of scripts that I use regularly, when I make a change and close the script. The change is persisted without asking if i wish to keep it. 

**Steps to re-produce taken**
- Open Dbeaver
- Open a script from the project
- Amend the script 
- Close script
- Reopen and the change persists

Previously i would be asked if i wanted to save the change. This no longer seems to happen, i can't be sure how long this has been happening as i've only just noticed. I've looked through the settings for any form of autosave and cannot seem to find any settings relating to it? 


**Dbeaver version**
Version 24.1.4.202408041450

**OS**
MacOS 14.5

**Comment by ask9** [comment, marked as answer]
Since DBeaver 24.1.1 the setting `Auto-save editor on close ` is turned on by default.
Check this thread for more information:
https://github.com/orgs/dbeaver/discussions/35158

**Comment by doughobbs** [comment]
> Since DBeaver 24.1.1 the setting `Auto-save editor on close ` is turned on by default. Check this thread for more information: https://github.com/orgs/dbeaver/discussions/35158

Many thanks for providing a solution.

Very curious to know why this was changed, seems a bit backwards to me! (especially as its been that way for a very long time).

**Comment by ask9** [comment]
> Very curious to know why this was changed, seems a bit backwards to me! (especially as its been that way for a very long time).

As I am just an ordinary user, I can only guess.
I think it can be benefitial in cases when a user has a lot of opened and unsaved tabs with scripts, and he wants to close the application (for example, to install the DBeaver's updates or reboot the system quickly) and doesn't want to accept confirmation about saving changes in 30 opened scripts 30 times.
The problem is that some accidential or unwanted changes can be automatically saved in scripts and user may not notice it or notice it too late.

Lets wait for official answer from DBeaver's team.

**Comment by doughobbs** [comment]
> > Very curious to know why this was changed, seems a bit backwards to me! (especially as its been that way for a very long time).
> 
> As I am just an ordinary user, I can only guess. I think it can be benefitial in cases when a user has a lot of opened and unsaved tabs with scripts, and he wants to close the application (for example, to install the DBeaver's updates or reboot the system quickly) and doesn't want to accept confirmation about saving changes in 30 opened scripts 30 times. The problem is that some accidential or unwanted changes can be automatically saved in scripts and user may not notice it or notice it too late.
> 
> Lets wait for official answer from DBeaver's team.

Yeah I can get that viewpoint and approach for those that use it very heavily like that, I only use it with a max of about 4 or 5 scripts open at once.

**Comment by asmith04** [comment]
> Since DBeaver 24.1.1 the setting `Auto-save editor on close ` is turned on by default. Check this thread for more information: https://github.com/orgs/dbeaver/discussions/35158

Ace, Thank you for that. I guess there is a rationale behind turning it on by default, I can definitely see some use cases. I didn't think to check the top level of that menu to be honest. I assumed it would be a sub entry. 

Most scripts i use are pulled from Devops / github and are template based so having the template overwritten can be awkward at times. Problem solved now though :)

**Comment by ask9** [comment]
> I didn't think to check the top level of that menu to be honest. I assumed it would be a sub entry.

When this issue will be done #34632  it will be much easier to find the certain option in Preferences.

**Comment by E1izabeth** [comment]
Lots of popular editors have saves file on close automatically. We received a request from the community (#29925) and decided to turn it on.
If you want it to be turned off by default again, you can vote for the issue #34514  with 👍 under the issue description.

I'm closing this issue as a question.

**Comment by BenjaminWKI** [comment]
Why was this change in default behaviour done without informing everyone?!? 
This is an incredibly dangerous assumption!
I nearly ended up destroying the only copies of my scripts when I was closing the editor instead of having to rollback thousands upon thousands of changes by holding Undo Ctrl-Z for several minutes.
Yes, popular editors like Notepad and VSCode do retain changes to a file when you close them, **but the key difference is that they save those changes to a temporary file and never overwrite the original file unless we explicitly save or confirm in a dialog that appears when we close the opened file!**
This "auto-save" is not the same and pulling something like this on unsuspecting users doesn't get you into the "cool & popular editors" club!
If I were to make changes to thesole copy of a script just to run tests ... but accidentaly have the program closed (like a crash or forced windows shutdown) there goes the sole copy of said file (along with the original file modified timestamp) and all the changes tracking to use Undo Ctrl-Z on!
**NEVER assume everyone wants to save changes to a file without their explicit consent!**
